src/SDL/Input/GameController.hs

```diff
@@ -57,14 +57,17 @@ instance FromNumber ControllerButton Int32 where
     _ -> ControllerButtonInvalid
 
 -- | Identifies the state of a controller button.
-data ControllerButtonState = ControllerButtonPressed | ControllerButtonReleased
+data ControllerButtonState
+  = ControllerButtonPressed
+  | ControllerButtonReleased
+  | ControllerButtonInvalidState
```

Comment:
Maybe "Unknown" rather than "Invalid" and pair up the `Word32` here?

Reply:
It's called `invalid` in the [SDL source code as well](http://hg.libsdl.org/SDL/file/084b3caa1bc8/src/joystick/SDL_gamecontroller.c#l1769), which is why I called it that way.